modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectRelationshipResourceImpl.java

```diff
@@ -123,6 +144,53 @@ public ObjectRelationship getObjectRelationship(Long objectRelationshipId)
 				objectRelationshipId));
 	}
 
+	@Override
+	public ObjectRelationship
+			postObjectDefinitionByExternalReferenceCodeObjectDefinitionExternalReferenceCodeObjectRelationship(
+				String objectDefinitionExternalReferenceCode,
+				ObjectRelationship objectRelationship)
+		throws Exception {
+
+		com.liferay.object.model.ObjectDefinition objectDefinition1 =
+			_objectDefinitionLocalService.
+				getObjectDefinitionByExternalReferenceCode(
+					objectDefinitionExternalReferenceCode,
+					contextCompany.getCompanyId());
+
+		com.liferay.object.model.ObjectDefinition objectDefinition2 =
+			_objectDefinitionLocalService.
+				getObjectDefinitionByExternalReferenceCode(
+					objectRelationship.
+						getObjectDefinitionExternalReferenceCode2(),
+					contextCompany.getCompanyId());
+
+		objectRelationship.setParameterObjectFieldId(
+			() -> {
+				if (Validator.isNotNull(
+						objectRelationship.getParameterObjectFieldName())) {
+
+					ObjectField objectField =
+						_objectFieldLocalService.getObjectField(
+							objectDefinition2.getObjectDefinitionId(),
+							objectRelationship.getParameterObjectFieldName());
+
+					return objectField.getObjectFieldId();
+				}
+
+				return 0L;
```

Comment:
@p-albuquerque can we buy some space here?

```
if (Validator.Null(
		objectRelationship.getParameterObjectFieldName())) {

	return 0L;
}

ObjectField objectField =
	_objectFieldLocalService.getObjectField(
		objectDefinition2.getObjectDefinitionId(),
		objectRelationship.getParameterObjectFieldName());

return objectField.getObjectFieldId();
```

Reply:
got it